### DISCUSSION TITLE
Need help with destroy space when empty script

### DISCUSSION THREAD

**Discussion by tbsmn** [asmvik/yabai#750] [Q&A] [answered]
I guess I have something wrong here but I can't see it. Maybe someone here could help me.

Objective: Empty spaces should be destroyed when switched to another space. (To cleanup unused spaces)

Here is my code for achieving this:
`yabai -m signal --add event=space_changed action='if [ $(yabai -m query --spaces --space $YABAI_RECENT_SPACE_ID | jq -re '.windows') = "[]" ]; then yabai -m space $YABAI_RECENT_SPACE_ID --destroy ; fi'  
`
This actually destroys empty spaces but it seems to trigger randomly.

Any Ideas?

**Comment by tbsmn** [comment, marked as answer]
I finaly figured it out myself. The problem was that spaces have an "id" and an independent "index":
If anyone is interessted here is the code that works:

`yabai -m signal --add event=space_changed action='RSINDEX=$(yabai -m query --spaces | jq ".[] | select (.id == $YABAI_RECENT_SPACE_ID) | .index") && if [ "$(yabai -m query --spaces --space $RSINDEX | jq -re '.windows')" = "[]" ] ; then yabai -m space $RSINDEX --destroy ; fi'`

**Reply by dominiklohmann** [reply]
Thanks for posting the solution as well. I converted this to discussions and marked this as the correct answer.

**Reply by Bellavene** [reply]
Hmm, suddenly stopped working
```
jq: error: syntax error, unexpected ')' (Unix shell quoting issues?) at <top-level>, line 1:
.[] | select (.id == ) | .index                     
jq: 1 compile error
```

**Reply by tbsmn** [reply]
I just looked it up. It seems like I have changed the script a while ago. But I can't remember why.

I now use this in my `.yabairc`:

`yabai -m signal --add event=space_changed action='~/scripts/destroy_empty_space.sh $YABAI_RECENT_SPACE_ID'`

And `destroy_empty_space.sh` looks like this:
```
#!/usr/bin/env bash

RSINDEX=$(yabai -m query --spaces | jq ".[] | select (.id == $1) | .index")
SPACEISEMPTY=$(yabai -m query --spaces --space $RSINDEX | jq -re '."first-window"')

if [[ $SPACEISEMPTY = 0 ]] ; then
    yabai -m space $RSINDEX --destroy
fi
```

Maybe this will work for you too.

**Comment by qns7** [comment]
Awesome, thank you for this! :)
